packages/db-states-redis/lib/states/statesInRedisClient.js

```diff
@@ -49,30 +46,28 @@ class StateRedis {
         this.client = null;
         this.sub = null;
         this.subSystem = null;
-        const ioRegExp = new RegExp('^' + this.namespaceRedis.replace(/\./g, '\\.') + '[_A-Za-z0-9ÄÖÜäöüа-яА-Я]+'); // io.[_A-Za-z0-9]+
 
-        this.log = this.settings.logger;
-        if (!this.log) {
-            this.log = {
-                silly: function (_msg) {/* console.log(msg); */
-                },
-                debug: function (_msg) {/* console.log(msg); */
-                },
-                info: function (_msg) {/* console.log(msg); */
-                },
-                warn: function (msg) {
-                    console.log(msg);
-                },
-                error: function (msg) {
-                    console.log(msg);
-                }
-            };
-        } else if (!this.log.silly) {
-            this.log.silly = this.log.debug;
+        this.log = tools.getLogger(this.settings.logger);
+
+        if (this.settings.autoConnect === undefined || this.settings.autoConnect) {
+            this.connectDb();
         }
+    }
 
+    connectDb() {
         this.settings.connection = this.settings.connection || {};
 
+        const onChange = this.settings.change; // on change handler
+        const onChangeUser = this.settings.changeUser; // on change handler for User events
+
+        const ioRegExp = new RegExp('^' + this.namespaceRedis.replace(/\./g, '\\.') + '[_A-Za-z0-9ÄÖÜäöüа-яА-Я]+'); // io.[_A-Za-z0-9]+
```

Comment:
might wanna use the new regex I added to js-controller here?

Reply:
puuh ... yes but adjusted because here it is about the "allowed first character" (so basically adapter name requirements) ... so not all are allowed it seems ... do we eed to adjust the other one too? So it is "number or character or underscore" allowed as first number ...e.g. no "-" or other characters that are normally.
So it is not the same ...

Reply:
Oh - yeah I guess we need to reflect that in the other regex aswell

Reply:
But careful ... in the other regex kind of only for "Foreign IDs" :-) because for "adapter-local" states it is as it is ...